server/service/hosts.go

```diff
@@ -20,32 +20,36 @@ import (
 // rendering in the UI.
 type HostResponse struct {
 	*fleet.Host
-	Status      fleet.HostStatus `json:"status"`
-	DisplayText string           `json:"display_text"`
-	Labels      []fleet.Label    `json:"labels,omitempty"`
+	Status      fleet.HostStatus   `json:"status"`
+	DisplayText string             `json:"display_text"`
+	Labels      []fleet.Label      `json:"labels,omitempty"`
+	Geolocation *fleet.GeoLocation `json:"geolocation,omitempty"`
 }
 
 func hostResponseForHost(ctx context.Context, svc fleet.Service, host *fleet.Host) (*HostResponse, error) {
 	return &HostResponse{
 		Host:        host,
 		Status:      host.Status(time.Now()),
 		DisplayText: host.Hostname,
+		Geolocation: svc.LookupGeoIP(ctx, host.PublicIP),
```

Comment:
@zwass in response to https://github.com/fleetdm/fleet/pull/4652#discussion_r829253005

to facilitate this geoip lookup in this host presenter function, without introducing new dependencies, I have `fleet.Service` implement `LookupIP`, and it just uses the underlying implementation injected at service creation time.

If it makes more sense to inject the dependency directly rather than exposing it on `fleet.Service` I can investigate that.

Reply:
This makes sense, I'll mark that as resolved.